test/util/conditional.spec.js

```diff
@@ -0,0 +1,231 @@
+        type: 'GreaterThanOrEqualTo',
+        object: '56629a41-b24b-42c6-9f02-277412e96d25'
+      }, {
+        id: '56629a41-b24b-42c6-9f02-277412e96d25'
+      })).to.equal(true);
+    });
+
+    it('Testing eval "Between"', () => {
+      expect(evaluate({
+        subject: 'id',
+        type: 'Between',
+        object: '56629a41-b24b-42c6-9f02-277412e96d25',
+        lowerBound: 1,
+        upperBound: 3
+      }, {
+        id: 2,
+        subject: 'id'
+      })).to.equal(true);
+    });
+
+    it('Testing eval "Membership"', () => {
+      expect(evaluate({
+        subject: 'id',
+        type: 'Membership',
+        object: '56629a41-b24b-42c6-9f02-277412e96d25',
+        values: ['id']
+      }, {
+        id: '56629a41-b24b-42c6-9f02-277412e96d25',
+        subject: 'id'
+      })).to.equal(false);
+    });
+
+    it('Testing eval "Not"', () => {
+      expect(evaluate({
+        subject: 'id',
+        type: 'Not',
+        object: '56629a41-b24b-42c6-9f02-277412e96d25',
+        condition: { type: 'Equals' }
+      }, {
+        id: '56629a41-b24b-42c6-9f02-277412e96d25'
+      })).to.equal(false);
+    });
+
+    it('Testing eval "And"', () => {
+      expect(evaluate({
+        subject: 'id',
+        type: 'And',
+        object: '56629a41-b24b-42c6-9f02-277412e96d25',
+        conditions: [{ type: 'Equals' }]
+      }, {
+        id: '56629a41-b24b-42c6-9f02-277412e96d25'
+      })).to.equal(true);
+    });
+
+    it('Testing eval "Or"', () => {
+      expect(evaluate({
+        subject: 'id',
+        type: 'Or',
+        object: '56629a41-b24b-42c6-9f02-277412e96d25',
+        conditions: [{ type: 'Equals' }]
+      }, {
+        id: '56629a41-b24b-42c6-9f02-277412e96d25'
+      })).to.equal(true);
+    });
+
+    it('Testing eval "Function"', (done) => {
+      try {
+        evaluate({
+          subject: 'id',
+          type: 'Function',
+          object: '56629a41-b24b-42c6-9f02-277412e96d25'
+        }, {
+          id: '56629a41-b24b-42c6-9f02-277412e96d25'
+        });
+      } catch (err) {
+        expect(err).instanceOf(ConditionNotImplemented);
+        expect(err.message).to.equal('Condition not implemented');
+        done();
+      }
+    });
+
+    it('Testing eval "Default"', (done) => {
+      try {
+        evaluate({
+          subject: 'id',
+          type: 'Default',
+          object: '56629a41-b24b-42c6-9f02-277412e96d25'
+        }, {
+          id: '56629a41-b24b-42c6-9f02-277412e96d25'
+        });
+      } catch (err) {
+        expect(err).instanceOf(UnknownConditionType);
+        expect(err.message).to.equal('Unknown condition type "Default" provided');
+        done();
+      }
+    });
+  });
+
+  describe('Testing extract', () => {
+    [
+      'Equals',
+      'NotEquals',
+      'LessThan',
+      'LessThanOrEqualTo',
+      'GreaterThan',
+      'GreaterThanOrEqualTo',
+      'Between',
+      'Membership'
+    ]
+      .forEach((c) => {
+        it(`Testing extract "${c}"`, () => {
```

Comment:
ehh this isn't it. We'll fix it in the next PR, but you don't need to have separate `it`s.
Think about what you're testing, it's not these specific conditions, it's that for this list, they extract the subject.
```js
.forEach((c) => {
  expect(..., `${c}`).to.deep.equal(...);
});
```